From 2433613c4e9a5a48cbec376da669d1ebd8ea7f18 Mon Sep 17 00:00:00 2001
From: Jim Bride <jim.bride@intel.com>
Date: Sun, 25 Feb 2018 10:41:38 -0800
Subject: [PATCH 1093/2328] drm/i915/gvt: Clean-up tiling initialization for
 gvt framebuffers

In the process of looking to fix a compiler warning, I noticed that
the way tiled framebuffers were being initialized wasn't correct,
so I fixed that and eliminated the warning.

Fixes:  906a56928786 ("drm/i915/gvt: Support vGPU guest framebuffer GEM object")
Signed-off-by: Jim Bride <jim.bride@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gvtbuffer.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gvtbuffer.c b/drivers/gpu/drm/i915/i915_gem_gvtbuffer.c
index d2939c2..05af0a0 100644
--- a/drivers/gpu/drm/i915/i915_gem_gvtbuffer.c
+++ b/drivers/gpu/drm/i915/i915_gem_gvtbuffer.c
@@ -254,21 +254,25 @@ i915_gem_gvtbuffer_ioctl(struct drm_device *dev, void *data,
 	}
 
 	if (IS_SKYLAKE(dev_priv) || IS_BROXTON(dev_priv)) {
-		unsigned int tiling_mode;
+		unsigned int tiling_mode = I915_TILING_NONE;
+		unsigned int stride = 0;
+
 		switch (args->tiled << 10) {
 		case PLANE_CTL_TILED_LINEAR:
-			tiling_mode = I915_TILING_NONE;
+			/* Default valid value */
 			break;
 		case PLANE_CTL_TILED_X:
 			tiling_mode = I915_TILING_X;
+			stride = args->stride;
 			break;
 		case PLANE_CTL_TILED_Y:
 			tiling_mode = I915_TILING_Y;
+			stride = args->stride;
 			break;
 		default:
 			DRM_ERROR("gvt: tiling mode %d not supported\n", args->tiled);
 		}
-		obj->tiling_and_stride = tiling_mode | args->stride;
+		obj->tiling_and_stride = tiling_mode | stride;
 	} else {
 		obj->tiling_and_stride = (args->tiled ? I915_TILING_X : I915_TILING_NONE) |
 			                 (args->tiled ? args->stride : 0);
-- 
2.7.4

